lib/fake_artist/table.ex

```diff
@@ -33,85 +53,209 @@ defmodule FakeArtist.Table do
+        players: players
+    }
 
-    {:reply, :ok, {id_map, table_name, user_count, state_pid}}
+    FakeArtistWeb.Endpoint.broadcast("table:#{table_name}", "update", state)
+
+    {:reply, :ok, state}
   end
 
-  def handle_call(:progress_game, _from, {id_map, table_name, user_count, state_pid}) do
-    %{seats: seats, active_seat: active_seat} = FakeArtist.Game.progress_game(state_pid)
-    players = assemble_players(seats, id_map, active_seat)
+  def handle_call(
+        :choose_category,
+        _from,
+        state = %{
+          active_players: active_players,
+          players: players,
+          table_name: table_name
+        }
+      ) do
+    state = %{
+      state
+      | active_players: get_active_players(players, active_players),
+        remaining_turns: ((players |> Map.keys() |> length()) - 1) * 2,
+        little_state: :draw
+    }
 
-    FakeArtistWeb.Endpoint.broadcast("table:#{table_name}", "update_game", %{players: players})
+    FakeArtistWeb.Endpoint.broadcast("table:#{table_name}", "update", state)
 
-    {:reply, :ok, {id_map, table_name, user_count, state_pid}}
+    {:reply, :ok, state}
   end
 
-  def handle_call(:choose_category, _from, {id_map, table_name, user_count, state_pid}) do
-    %{seats: seats, active_seat: active_seat} = FakeArtist.Game.choose_category(state_pid)
-    players = assemble_players(seats, id_map, active_seat)
+  def handle_call(
+        :progress_game,
+        _from,
+        state = %{
+          active_players: active_players,
+          players: players,
+          remaining_turns: remaining_turns,
+          table_name: table_name
+        }
+      ) do
+    if remaining_turns == 1 do
+      Logger.info(fn -> "Voting." end)
+
+      state = %{
+        state
+        | little_state: :vote
+      }
+
+      {:reply, :ok, state}
+    else
+      state = %{
+        state
+        | active_players: get_active_players(players, active_players),
+          remaining_turns: remaining_turns - 1
+      }
 
-    FakeArtistWeb.Endpoint.broadcast("table:#{table_name}", "update_game", %{players: players})
+      FakeArtistWeb.Endpoint.broadcast("table:#{table_name}", "update", state)
 
-    {:reply, :ok, {id_map, table_name, user_count, state_pid}}
+      {:reply, :ok, state}
+    end
   end
 
   def handle_info(
         {:DOWN, _ref, :process, _from, _reason},
-        {id_map, table_name, user_count, _old_state_pid}
+        state = %{connected_computers: connected_computers}
       ) do
     Logger.info(fn -> "table lost connection." end)
-    Logger.info(fn -> "player count decreased from #{user_count} to #{user_count - 1}" end)
 
-    user_count = user_count - 1
+    Logger.info(fn ->
+      "player count decreased from #{connected_computers} to #{connected_computers - 1}"
+    end)
 
-    if user_count == 0 do
+    if connected_computers - 1 == 0 do
       Logger.info(fn -> "suicide" end)
-      {:stop, :shutdown, {%{}, "", 0}}
+      {:stop, :shutdown, %{}}
```

Comment:
The 3rd arg here should be the final state of the server before it terminates so you can maybe use it in the [`terminate/2`](https://hexdocs.pm/elixir/GenServer.html#c:terminate/2) callback.

We're not defining `terminate/2` yet but we might in the future, and in either case we should probably pass the state instead of an empty Map